tests/test_data_bleach.py

```diff
@@ -0,0 +1,70 @@
+import json
+import os
+import unittest
+import uuid
+from pathlib import Path
+
+from scripts.data_bleach import main, PUBLIC_FIELDS
+
+
+class TestDataBleach(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.test_path = Path(__file__).parents[0]
+        cls.test_file = cls.test_path / 'test_data/bleach_test_data.json'
+        cls.out_file = cls.test_path / f'{str(uuid.uuid4())}.tmp.json'
+
+    def setUp(self):
+        with self.assertRaises(FileNotFoundError):
+            with open(self.out_file, 'r'):
+                pass
+
+    def tearDown(self):
+        try:
+            os.remove(self.out_file)
+        except FileNotFoundError:
+            pass
+
+    def _get_output_json(self):
+        with open(self.out_file, 'r') as fp:
+            return json.load(fp)
+
+    @staticmethod
+    def _field_is_public(field, sub_field):
+        for public_field, public_sub_field in PUBLIC_FIELDS:
+            if field == public_field:
+                if sub_field == public_sub_field:
+                    return True
+        return False
+
+    def _properly_sanitized(self, metadata):
+        for field in metadata:
+            if type(metadata[field]) != dict:
+                # the field should be cleared (sanitized)
+                assert not metadata[field]
+            else:
+                for sub_field in metadata[field]:
+                    if self._field_is_public(field, sub_field):
+                        # public fields have data
+                        assert metadata[field][sub_field]
```

Comment:
What about the case where field is public but the value is undefined (`null`, `None`)?
It seems that is a common case that we probably should include in this unit test, and if we had test input data which contained public fields that were `null`/`None` this line in `_properly_sanitized` would fail inappropriately.

Reply:
True, I didn't think about that. Are there any reasonable assertions we can make about these fields? I'm leaning towards no...

Reply:
There are reasonable assertions you can/should make if you know your test input data, which you do. You could have one test input in which the values are defined, and another test input in which the values are undefined.